src/unix/fs.c

```diff
@@ -421,6 +488,116 @@ static ssize_t uv__fs_scandir(uv_fs_t* req) {
   return n;
 }
 
+static int uv__fs_opendir(uv_fs_t* req) {
+  DIR* dir_stream;
+  uv__dirent_t *dirent;
+  uv_dir_t *dir;
+  size_t len;
+
+  assert(req);
+
+  dir_stream = opendir(req->path);
+  if (dir_stream == NULL)
+    return -1;
+
+  len = uv__fs_direntry_size(dir_stream);
```

Comment:
I see you still use `uv__fs_direntry_size` in order to malloc `dir->dirent` but it seems like it is unused now. Is that needed at all now that [readdir(3)](http://man7.org/linux/man-pages/man3/readdir.3.html) is returning the entries?